libsolutil/CommonIO.h

```diff
@@ -39,7 +39,7 @@ namespace solidity::util
 std::string readFileAsString(std::string const& _file);
 
 /// Retrieves and returns the whole content of the specified input stream (until EOF).
-std::string readUntilEnd(std::istream& _stdin);
+std::string readUntilEnd(std::istream const& _stdin);
```

Comment:
Interesting that it actually compiles with `const` here. Probably because by accessing `rdbuf()` you're not marking the text you read a actually read. I wonder if it might be better to flush the stream so that other parts of the compiler cannot accidentally process the same input again.

Reply:
It should not be `const` anyhow. Otherwise the caller might think it's actually not consuming from `_stdin`?

Reply:
But it's not consuming stdin content :) I think you can still read it after calling this function because `rdbuf()` just gives you the internal pointer to the content of the stream.

The question is just whether it's a good thing or not. I think that currently the name implies that the content would be consumed so I think I'd be in favor of making it non-`const` and making it clear the buffer.

Reply:
`peakUntilEnd()`?

Reply:
I think that would be `peekUntilEnd()`. But other than that it sounds good to me. I originally wanted to call it `captureInputStream()` when I refactored it recently. I like "peek" though because it would immediately tell me that the function does not consume stuff from the stream.

Reply:
@cameel also, "peek" is a common term in text processing and known not to consume (as opposed to "get", "read", "consume", etc.)

Reply:
Yeah. That's what I meant by saying that it tells me immediately what actually happens :)

Reply:
@cameel Another concern I have: if we keep `readUntilEnd(istream&)`, then it should definitely consume the input (we can still add a `peekUntilEnd()`.

Reply:
Is there any situation where we would want this function to consume the input?

Reply:
Looks like outputting the `rdbuf()` does advance the stream (the function is `const` but the pointer it returns is not) so I changed the parameter back to a non-`const` reference and renamed the function back to `readUntilEnd()`.